src/lib.rs

```diff
@@ -76,6 +76,31 @@ pub struct FrameInfo {
     pub component_count: i32,
 }
 
+
+/// Interleave modes for encoding/decoding.
+///
+/// Defines how color components are arranged in memory for both native and encoded formats.
+/// See more: https://github.com/team-charls/charls/wiki/How-to-use-the-API#multi-component-image-typical-rgb-color
+#[derive(Debug, Default)]
```

Comment:
Other useful traits which can be derived here: `Clone`, `Copy`, `PartialEq`, `Eq`, `Hash`